Invalid Mime-type? Cannot upload customers files with new core

The file I was trying to upload was downloaded from a viur2 production system, it is an *.oft file. I don't know what the correct mimetype should be here, but I doubt that we are doing everything correctly when checking. Why do we check the mimetype at all? Do we reject any type of file or is this just for the mimetype field?

```
[2023-04-27 13:44:13,608] /.../viur/core/request.py:299 [ERROR] ViUR has caught an unhandled exception!
[2023-04-27 13:44:13,608] /.../viur/core/request.py:300 [ERROR] Invalid Mime-Type
Traceback (most recent call last):
  File "/home/user/.local/share/virtualenvs/project-viur3-5lETOfwa/lib/python3.10/site-packages/viur/core/request.py", line 275, in processRequest
    self.findAndCall(path)
  File "/home/user/.local/share/virtualenvs/project-viur3-5lETOfwa/lib/python3.10/site-packages/viur/core/request.py", line 623, in findAndCall
    res = caller(*newArgs, **newKwargs)
  File "/home/user/.local/share/virtualenvs/project-viur3-5lETOfwa/lib/python3.10/site-packages/viur/core/modules/file.py", line 520, in getUploadURL
    assert len(mimeType.split("/")) == 2, "Invalid Mime-Type"
AssertionError: Invalid Mime-Type
[2023-04-27 13:44:13] [500] [POST] /json/file/getUploadURL
```

--- a/core/bones/base.py
+++ b/core/bones/base.py
@@ -16,25 +16,25 @@ from viur.core import db
 from viur.core.config import conf
 
 
-__systemIsIntitialized_ = False
+__system_initialized = False
 """
-Initializes the global variable __systemIsIntitialized
+Initializes the global variable __system_initialized
 """
 
 
 def setSystemInitialized():
     """
-    Sets the global __systemIsIntitialized_ variable to True, indicating that the system is
+    Sets the global __system_initialized variable to True, indicating that the system is
     initialized and ready for use. This function should be called once all necessary setup
     tasks have been completed. It also iterates over all skeleton classes and calls their
     setSystemInitialized() method.
 
     Global variables:
-        __systemIsIntitialized_: A boolean flag indicating if the system is initialized.
+        __system_initialized: A boolean flag indicating if the system is initialized.
     """
-    global __systemIsIntitialized_
+    global __system_initialized
     from viur.core.skeleton import iterAllSkelClasses
-    __systemIsIntitialized_ = True
+    __system_initialized = True
     for skelCls in iterAllSkelClasses():
         skelCls.setSystemInitialized()
 
@@ -42,10 +42,10 @@ def setSystemInitialized():
 def getSystemInitialized():
     """
     Retrieves the current state of the system initialization by returning the value of the
-    global variable __systemIsIntitialized_.
+    global variable __system_initialized.
     """
-    global __systemIsIntitialized_
-    return __systemIsIntitialized_
+    global __system_initialized
+    return __system_initialized
 
 
 class ReadFromClientErrorSeverity(Enum):

--- a/core/modules/file.py
+++ b/core/modules/file.py
@@ -518,7 +518,7 @@ class File(Tree):
         authSig = kwargs.get("authSig")
 
         # Validate the contentType from the client seems legit
-        mimeType = mimeType.lower()
+        mimeType = mimeType.strip().lower()
         if not (
             mimeType
             and mimeType.count("/") == 1

--- a/core/modules/file.py
+++ b/core/modules/file.py
@@ -520,7 +520,8 @@ class File(Tree):
         # Validate the contentType from the client seems legit
         mimeType = mimeType.lower()
         if not (
-            len(mimeType.split("/")) == 2
+            mimeType
+            and mimeType.count("/") == 1
             and all([ch in string.ascii_letters + string.digits + "/-.+" for ch in mimeType])
         ):
             raise errors.NotAcceptable("Invalid mimeType provided")

--- a/core/modules/file.py
+++ b/core/modules/file.py
@@ -524,7 +524,7 @@ class File(Tree):
             and mimeType.count("/") == 1
             and all(ch in string.ascii_letters + string.digits + "/-.+" for ch in mimeType)
         ):
-            raise errors.UnprocessableEntity(f"Invalid mimeType {mimeType} provided")
+            raise errors.UnprocessableEntity(f"Invalid mime-type {mimeType} provided")
 
         if authData and authSig:
             # First, validate the signature, otherwise we don't need to proceed further
@@ -542,7 +542,7 @@ class File(Tree):
                         validMimeType.endswith("*") and mimeType.startswith(validMimeType[:-1])):
                         break
                 else:
-                    raise errors.UnprocessableEntity(f"Invalid mimeType {mimeType} provided")
+                    raise errors.UnprocessableEntity(f"Invalid mime-type {mimeType} provided")
 
             node = authData["node"]
             maxSize = authData["maxSize"]

--- a/core/modules/file.py
+++ b/core/modules/file.py
@@ -512,21 +512,29 @@ class File(Tree):
         return db.encodeKey(fileSkel["key"]), uploadUrl
 
     @exposed
-    def getUploadURL(self, fileName, mimeType, size=None, skey=None, *args, **kwargs):
+    def getUploadURL(self, fileName: str, mimeType: str, size: int = None, skey: str = None, *args, **kwargs):
         node = kwargs.get("node")
         authData = kwargs.get("authData")
         authSig = kwargs.get("authSig")
-        # Validate the the contentType from the client seems legit
+
+        # Validate the contentType from the client seems legit
         mimeType = mimeType.lower()
-        assert len(mimeType.split("/")) == 2, "Invalid Mime-Type"
-        assert all([x in string.ascii_letters + string.digits + "/-.+" for x in mimeType]), "Invalid Mime-Type"
+        if not (
+            len(mimeType.split("/")) == 2
+            and all([ch in string.ascii_letters + string.digits + "/-.+" for ch in mimeType])
+        ):
+            raise errors.NotAcceptable("Invalid mimeType provided")
+
         if authData and authSig:
-            # First, validate the signature, otherwise we don't need to proceed any further
+            # First, validate the signature, otherwise we don't need to proceed further
             if not utils.hmacVerify(authData.encode("ASCII"), authSig):
-                raise errors.Forbidden()
+                raise errors.Unauthorized()
+
             authData = json.loads(base64.b64decode(authData.encode("ASCII")).decode("UTF-8"))
+
             if datetime.strptime(authData["validUntil"], "%Y%m%d%H%M") < datetime.now():
                 raise errors.Gone()
+
             if authData["validMimeTypes"]:
                 for validMimeType in authData["validMimeTypes"]:
                     if validMimeType == mimeType or (
@@ -534,8 +542,10 @@ class File(Tree):
                         break
                 else:
                     raise errors.NotAcceptable()
+
             node = authData["node"]
             maxSize = authData["maxSize"]
+
         else:
             if node:
                 rootNode = self.getRootNode(node)
@@ -544,25 +554,25 @@ class File(Tree):
             else:
                 if not self.canAdd("leaf", None):
                     raise errors.Forbidden()
+
             maxSize = None  # The user has some file/add permissions, don't restrict fileSize
+
         if maxSize:
-            try:
-                size = int(size)
-                assert size <= maxSize
-            except:  # We have a size-limit set - but no size supplied
-                raise errors.PreconditionFailed()
+            if size > maxSize:
+                raise errors.Forbidden("Size limit exceeds maximum size")
         else:
             size = None
 
         if not securitykey.validate(skey, useSessionKey=True):
             raise errors.PreconditionFailed()
 
-        targetKey, uploadUrl = self.initializeUpload(fileName, mimeType.lower(), node, size)
+        targetKey, uploadUrl = self.initializeUpload(fileName, mimeType, node, size)
 
         resDict = {
             "uploadUrl": uploadUrl,
             "uploadKey": targetKey,
         }
+
         if authData and authSig:
             # In this case, we'd have to store the key in the users session so he can call add() later on
             session = current.session.get()
@@ -572,6 +582,7 @@ class File(Tree):
             # Clamp to the latest 50 pending uploads
             session["pendingFileUploadKeys"] = session["pendingFileUploadKeys"][-50:]
             session.markChanged()
+
         return self.render.view(resDict)
 
     @exposed

